### PR TITLE
wix-ui-core - migrate to @stylable/dom-test-kit

### PR DESCRIPTION
**Pull request by yardnsm**

### CODE REVIEW

File: packages/wix-ui-core/package.json [hunk 1]
```diff
@@ -60,6 +60,7 @@
   "devDependencies": {
     "@storybook/addon-options": "4.0.0-alpha.14",
     "@storybook/react": "4.0.0-alpha.14",
+    "@stylable/dom-test-kit": "^0.1.0",
```
Comment (shlomitc) [inline packages/wix-ui-core/package.json]: should be dependency

File: packages/wix-ui-core/src/globals.d.ts [hunk 2]
```diff
@@ -1,43 +1,32 @@
-declare module '*.st.css' {
-  const stylesheet: RuntimeStylesheet;
-  export default stylesheet;
-}
+type StateValue = boolean | number | string;
```
Comment (shlomitc) [inline packages/wix-ui-core/src/globals.d.ts]: changing this file might problematic.
Other componet libraries use it in their `tsconfig.json` and rely on the data.
for example:
https://github.com/wix/wix-ui-backoffice/blob/master/tsconfig.json#L17

can you make `wix-ui-backoffice` will work with this new change?
also - was this file change required when migrating to the test-utils or you just did it on the way?

Comment (yardnsm) [reply packages/wix-ui-core/src/globals.d.ts]: Yeah, this change was required when migrating to the new test-utils. @tomrav helped me with it.

Also, there is a related PR in `wix-ui-backoffice`: https://github.com/wix/wix-ui-backoffice/pull/213. These changes are required for it to pass as well.

File: packages/wix-ui-core/src/types/BaseProps.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /**
  * Base props for any core component
  */
-export interface BaseProps extends RuntimeStylesheetBaseProps {
+export interface BaseProps extends InheritedAttributes {
```
Comment (hepiyellow) [inline packages/wix-ui-core/src/types/BaseProps.ts]: @yardnsm Why did you change this?
It opens all component's Props type to any kind of prop.
```
interface InheritedAttributes {
    className?: string;
    [props: string]: any;
}
```
